regression: jbang alias no longer work with GAV refs

```
jbang run@jruby -e "puts 'Hello Ruby World'"
[jbang] [ERROR] Could not download https://raw.githubusercontent.com/jruby/jbang-catalog/master/org.jruby:jruby-complete:9.2.12.0
[jbang] Run with --verbose for more details
```

looks like the relative handling we added broke jbangs handling of GAV's refs in alias catalogs.

--- a/src/main/java/dev/jbang/AliasUtil.java
+++ b/src/main/java/dev/jbang/AliasUtil.java
@@ -634,7 +634,7 @@ public class AliasUtil {
 	}
 
 	private static boolean isRemoteRef(String ref) {
-		return ref.startsWith("http:") || ref.startsWith("https:");
+		return ref.startsWith("http:") || ref.startsWith("https:") || DependencyUtil.looksLikeAGav(ref);
 	}
 
 	@SafeVarargs

--- a/src/test/java/dev/jbang/cli/TestAlias.java
+++ b/src/test/java/dev/jbang/cli/TestAlias.java
@@ -51,6 +51,9 @@ public class TestAlias {
 			"    },\n" +
 			"    \"eight\": {\n" +
 			"      \"script-ref\": \"seven\"\n" +
+			"    },\n" +
+			"    \"gav\": {\n" +
+			"      \"script-ref\": \"org.example:artifact:version\"\n" +
 			"    }\n" +
 			"  }\n" +
 			"}";
@@ -92,6 +95,7 @@ public class TestAlias {
 		AliasUtil.Alias alias = AliasUtil.getAlias(cwd, "one", null, null);
 		assertThat(alias, notNullValue());
 		assertThat(alias.scriptRef, equalTo("http://dummy"));
+		assertThat(alias.resolve(cwd), equalTo("http://dummy"));
 		assertThat(alias.arguments, nullValue());
 		assertThat(alias.properties, nullValue());
 	}
@@ -102,6 +106,7 @@ public class TestAlias {
 				Collections.singletonMap("foo", "bar"));
 		assertThat(alias, notNullValue());
 		assertThat(alias.scriptRef, equalTo("http://dummy"));
+		assertThat(alias.resolve(cwd), equalTo("http://dummy"));
 		assertThat(alias.arguments, iterableWithSize(1));
 		assertThat(alias.arguments, contains("X"));
 		assertThat(alias.properties, aMapWithSize(1));
@@ -113,6 +118,7 @@ public class TestAlias {
 		AliasUtil.Alias alias = AliasUtil.getAlias(cwd, "two", null, null);
 		assertThat(alias, notNullValue());
 		assertThat(alias.scriptRef, equalTo("http://dummy"));
+		assertThat(alias.resolve(cwd), equalTo("http://dummy"));
 		assertThat(alias.arguments, iterableWithSize(1));
 		assertThat(alias.arguments, contains("2"));
 		assertThat(alias.properties, aMapWithSize(1));
@@ -124,6 +130,7 @@ public class TestAlias {
 		AliasUtil.Alias alias = AliasUtil.getAlias(cwd, "two", Collections.emptyList(), Collections.emptyMap());
 		assertThat(alias, notNullValue());
 		assertThat(alias.scriptRef, equalTo("http://dummy"));
+		assertThat(alias.resolve(cwd), equalTo("http://dummy"));
 		assertThat(alias.arguments, iterableWithSize(1));
 		assertThat(alias.arguments, contains("2"));
 		assertThat(alias.properties, aMapWithSize(1));
@@ -136,6 +143,7 @@ public class TestAlias {
 				Collections.singletonMap("foo", "bar"));
 		assertThat(alias, notNullValue());
 		assertThat(alias.scriptRef, equalTo("http://dummy"));
+		assertThat(alias.resolve(cwd), equalTo("http://dummy"));
 		assertThat(alias.arguments, iterableWithSize(1));
 		assertThat(alias.arguments, contains("X"));
 		assertThat(alias.properties, aMapWithSize(1));
@@ -147,6 +155,7 @@ public class TestAlias {
 		AliasUtil.Alias alias = AliasUtil.getAlias(cwd, "four", null, null);
 		assertThat(alias, notNullValue());
 		assertThat(alias.scriptRef, equalTo("http://dummy"));
+		assertThat(alias.resolve(cwd), equalTo("http://dummy"));
 		assertThat(alias.arguments, iterableWithSize(1));
 		assertThat(alias.arguments, contains("4"));
 		assertThat(alias.properties, aMapWithSize(1));
@@ -159,6 +168,7 @@ public class TestAlias {
 				Collections.singletonMap("foo", "bar"));
 		assertThat(alias, notNullValue());
 		assertThat(alias.scriptRef, equalTo("http://dummy"));
+		assertThat(alias.resolve(cwd), equalTo("http://dummy"));
 		assertThat(alias.arguments, iterableWithSize(1));
 		assertThat(alias.arguments, contains("X"));
 		assertThat(alias.properties, aMapWithSize(1));
@@ -170,6 +180,7 @@ public class TestAlias {
 		AliasUtil.Alias alias = AliasUtil.getAlias(cwd, "five", null, null);
 		assertThat(alias, notNullValue());
 		assertThat(alias.scriptRef, equalTo("http://dummy"));
+		assertThat(alias.resolve(cwd), equalTo("http://dummy"));
 		assertThat(alias.arguments, iterableWithSize(1));
 		assertThat(alias.arguments, contains("3"));
 		assertThat(alias.properties, aMapWithSize(1));
@@ -182,10 +193,21 @@ public class TestAlias {
 				Collections.singletonMap("foo", "bar"));
 		assertThat(alias, notNullValue());
 		assertThat(alias.scriptRef, equalTo("http://dummy"));
+		assertThat(alias.resolve(cwd), equalTo("http://dummy"));
 		assertThat(alias.arguments, iterableWithSize(1));
 		assertThat(alias.arguments, contains("X"));
 		assertThat(alias.properties, aMapWithSize(1));
 		assertThat(alias.properties, hasEntry("foo", "bar"));
+	}
+
+	@Test
+	void testGetAliasGav() throws IOException {
+		AliasUtil.Alias alias = AliasUtil.getAlias(cwd, "gav", null, null);
+		assertThat(alias, notNullValue());
+		assertThat(alias.scriptRef, equalTo("org.example:artifact:version"));
+		assertThat(alias.resolve(cwd), equalTo("org.example:artifact:version"));
+		assertThat(alias.arguments, nullValue());
+		assertThat(alias.properties, nullValue());
 	}
 
 	@Test

--- a/src/test/java/dev/jbang/cli/TestAliasWithBaseRef.java
+++ b/src/test/java/dev/jbang/cli/TestAliasWithBaseRef.java
@@ -1,0 +1,90 @@
+package dev.jbang.cli;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.*;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+
+import org.junit.Rule;
+import org.junit.contrib.java.lang.system.EnvironmentVariables;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.rules.TemporaryFolder;
+
+import dev.jbang.AliasUtil;
+
+public class TestAliasWithBaseRef {
+
+	static final String aliases = "{\n" +
+			"  \"base-ref\": \"http://dummy\",\n" +
+			"  \"aliases\": {\n" +
+			"    \"one\": {\n" +
+			"      \"script-ref\": \"foo\"\n" +
+			"    },\n" +
+			"    \"two\": {\n" +
+			"      \"script-ref\": \"foo/bar.java\"\n" +
+			"    },\n" +
+			"    \"three\": {\n" +
+			"      \"script-ref\": \"http://dummy/baz.java\"\n" +
+			"    },\n" +
+			"    \"gav\": {\n" +
+			"      \"script-ref\": \"org.example:artifact:version\"\n" +
+			"    }\n" +
+			"  }\n" +
+			"}";
+
+	@BeforeEach
+	void init() throws IOException {
+		jbangTempDir.create();
+		testTempDir.create();
+		environmentVariables.set("JBANG_DIR", jbangTempDir.getRoot().getPath());
+		Files.write(jbangTempDir.getRoot().toPath().resolve(AliasUtil.JBANG_CATALOG_JSON), aliases.getBytes());
+		cwd = testTempDir.newFolder("test").toPath();
+	}
+
+	@Rule
+	public final EnvironmentVariables environmentVariables = new EnvironmentVariables();
+
+	@Rule
+	public final TemporaryFolder jbangTempDir = new TemporaryFolder();
+
+	@Rule
+	public final TemporaryFolder testTempDir = new TemporaryFolder();
+
+	private Path cwd;
+
+	@Test
+	void testGetAliasOne() throws IOException {
+		AliasUtil.Alias alias = AliasUtil.getAlias(cwd, "one", null, null);
+		assertThat(alias, notNullValue());
+		assertThat(alias.scriptRef, equalTo("foo"));
+		assertThat(alias.resolve(cwd), equalTo("http://dummy/foo"));
+	}
+
+	@Test
+	void testGetAliasTwo() throws IOException {
+		AliasUtil.Alias alias = AliasUtil.getAlias(cwd, "two", null, null);
+		assertThat(alias, notNullValue());
+		assertThat(alias.scriptRef, equalTo("foo/bar.java"));
+		assertThat(alias.resolve(cwd), equalTo("http://dummy/foo/bar.java"));
+	}
+
+	@Test
+	void testGetAliasThree() throws IOException {
+		AliasUtil.Alias alias = AliasUtil.getAlias(cwd, "three", null, null);
+		assertThat(alias, notNullValue());
+		assertThat(alias.scriptRef, equalTo("http://dummy/baz.java"));
+		assertThat(alias.resolve(cwd), equalTo("http://dummy/baz.java"));
+	}
+
+	@Test
+	void testGetAliasGav() throws IOException {
+		AliasUtil.Alias alias = AliasUtil.getAlias(cwd, "gav", null, null);
+		assertThat(alias, notNullValue());
+		assertThat(alias.scriptRef, equalTo("org.example:artifact:version"));
+		assertThat(alias.resolve(cwd), equalTo("org.example:artifact:version"));
+	}
+
+}

